src/main/java/com/ericsson/gerrit/plugins/eiffel/handlers/DataBaseHandler.java

```diff
@@ -0,0 +1,245 @@
+     * @param filename
+     * @throws ConnectException
+     */
+    public DataBaseHandler(final File pluginDir, final String filename) throws ConnectException {
+        this(pluginDir.toString(), filename);
+    }
+
+    /**
+     * This function returns an event id if exists for a specific table depending on
+     * the keyValue
+     *
+     * @param table
+     * @param keyValue
+     * @return eventId
+     * @throws ConnectException
+     * @throws NoSuchElementException
+     */
+    public String getEventID(final Table table, final String keyValue) throws ConnectException, NoSuchElementException {
+        String eventID = "";
+
+        String sqlSelectStatement = String.format("SELECT * FROM %s WHERE %s=?", table, table.keyName);
+        try (Connection connection = connect();
+                PreparedStatement preparedStatement = connection.prepareStatement(sqlSelectStatement)) {
+
+            preparedStatement.setString(1, keyValue);
+            eventID = executeQuery(preparedStatement);
+
+        } catch (SQLException e) {
+            LOGGER.error("Error when trying to fetch values from database: {}\n{}", e.getMessage(), e);
+        }
+
+        if (eventID.isEmpty()) {
+            throw new NoSuchElementException("Database did not return any value for this query");
+        }
+
+        return eventID;
+    }
+
+    /**
+     * This function updates value to the given table. The keyValue value is
+     * different depending on Table (branch name for scs and change-id for scc)
+     *
+     * @param table
+     * @param keyValue
+     * @param eiffelevent
+     * @throws ConnectException
+     * @throws SQLException
+     */
+    public void updateInto(final Table table, final String keyValue, final String eiffelevent)
+            throws ConnectException, SQLException {
+        String sqlUpdateStatement = String.format("UPDATE %s SET %s=? WHERE %s=?", table, EVENT_ID_KEY, table.keyName);
+        executeUpdate(sqlUpdateStatement, keyValue, eiffelevent);
+    }
+
+    /**
+     * This function inserts values to the given table. The keyValue value is
+     * different depending on Table (branch name for scs and change-id for scc)
+     *
+     * @param table
+     * @param keyValue
+     * @param eiffelevent
+     * @throws SQLException
+     * @throws ConnectException
+     */
+    public void insertInto(final Table table, final String keyValue, final String eiffelevent)
+            throws SQLException, ConnectException {
+        String sqlInsertStatement = String.format("INSERT INTO %s(%s,%s) VALUES(?,?)", table, EVENT_ID_KEY,
+                table.keyName);
+        executeUpdate(sqlInsertStatement, keyValue, eiffelevent);
+
+    }
+
+    /**
+     * This function will create a sql database structure in the database file if it
+     * does not exist.
+     *
+     * @throws ConnectException
+     */
+    private void createNewDatabase() throws ConnectException {
+        try (Connection sqlConnection = connect()) {
```

Comment:
are we really creating a DB here?
I just see 2 calls one to close the connection and 2 to get the metadata. 
Is it not the connect() which actually creates DB?

Reply:
But that should be explained

Reply:
Added a comment for that.

Reply:
Ok for me